webapp/app/js/models/resultsets_store.js

```diff
@@ -822,15 +801,163 @@ treeherder.factory('ThResultSetModel', [
+    */
+    var groupJobByPlatform = function(jobList){
+        var groupedJobs = {
+            platforms:[],
+            job_counts: getJobCount(jobList)
+        };
+
+        if(jobList.length == 0){return groupedJobs;}
+        groupedJobs.id = jobList[0].result_set_id;
+        var lastModified = ""
+        for(var i=0; i<jobList.length; i++){
+            // search for the right platform
+            var job = jobList[i];
+            var platform = _.find(groupedJobs.platforms, function(platform){
+                return job.build_platform == platform.name &&
+                 job.platform_option == platform.option;
+            })
+            if(_.isUndefined(platform)){
+                platform = {
+                    name: job.build_platform,
+                    option: job.platform_option,
+                    groups: []
+                };
+                groupedJobs.platforms.push(platform);
+            }
+            // search for the right group
+            var group = _.find(platform.groups, function(group){
+                return job.job_group_name == group.name &&
+                job.job_group_symbol == group.symbol;
+            })
+            if(_.isUndefined(group)){
+                group = {
+                    name: job.job_group_name,
+                    symbol: job.job_group_symbol,
+                    jobs: []
+                };
+                platform.groups.push(group);
+            }
+            group.jobs.push(job);
+        }
+        groupedJobs.lastModified = lastModified;
+        return groupedJobs;
+    };
+
+    var jobTypeOrderPromise = ThJobTypeModel.get_list().
+        then(function(jobTypeList){
+            var jobSymbolList = _.uniq(_.pluck(jobTypeList, 'symbol'));
+            /*
+            * Symbol could be something like 1, 2 or 3. Or A, B, C or R1, R2, R10.
+            * So this will pad the numeric portion with 0s like R001, R010, etc.
+            */
+            var paddedJobSymbolList = _.map(
+                jobSymbolList, function(jobSymbol){
+                    return jobSymbol.replace(
+                        /([\D]*)([\d]*)/g,
+                        function(matcher, s1, s2){
+                            if(s2 !== ""){
+                                s2 = "00" + s2;
+                                s2 = s2.slice(-3)
+                                return s1 + s2
+                            }
+                            return matcher;
+                        }
+                    );
+                }
+            );
+            var symbolMap = _.object(paddedJobSymbolList, jobSymbolList);
+            paddedJobSymbolList.sort();
+            var outputOrderMap = {};
+            _.each(paddedJobSymbolList, function(paddedJob, index){
+                outputOrderMap[symbolMap[paddedJob]] = index;
+            })
+            return outputOrderMap;
+        });
+    var jobGroupOrderPromise = ThJobGroupModel.get_list().
```

Comment:
I'm curious here about the choice to fetch the whole list of ``job_groups`` and ``job_types`` rather than just sorting based on the ones you find in the result.  Does it make it harder to add new ones as new jobs come in?  We may hit the case where a new job type or group is added while the page is just updating.  So the cached list of these items may mean we don't see them until a page refresh?  Though maybe I'm missing something there.  :)